scripts/generate/templates/docs/usage/configuration/sources/docker.md.erb

```diff
@@ -0,0 +1,31 @@
+<%- component = metadata.sources.docker -%>
+
+<%= component_header(component) %>
+
+## Config File
+
+<%= component_config_example(component) %>
+
+## Options
+
+<%= options_table(component.options.to_h.values.sort) %>
+
+## How It Works [[sort]]
+
+<%= component_sections(component) %>
+
+### Connecting to the Docker daemon
+
+Vector will automatically attempt to connect to the docker daemon for you. In most
+situations if your current user is able to run `docker ps` then Vector will be able to
+connect. Vector will also respect if `DOCKER_HOST` and `DOCKER_VERIFY_TLS` are set. Vector will also
+use the other default docker environment variables if they are set. For more infomration on how to
+connect to the docker daemon can be found here https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-socket-option.
```

Comment:
Can we avoid long raw links like this? We also try to avoid saying "here". Ex:

```md
See the [Docker daemon docs][urls.docker_daemon] for more information on connecting to Docker.
```

Reply:
Yeah I see that makes sense.